.github/pull_request_template.md

```diff
@@ -0,0 +1,28 @@
+<!--
+    If you are changing an existing module or resource, describe why that configuration is no longer acceptable. If you are fixing a misconfiguration, make sure to describe both the intended result and the actual (faulty) result.
+-->
+
+<!--
+    If you are adding a module or resources, describe what usage will be supported when this change is applied.
+-->
+
+<!--
+    Explain what alternative configurations you have evaluated and why the chosen configuration is preferable.
+-->
+
+<!--
+    What do reviewers need to know to review? Link pull requests or issues that are related.
+-->
+
+<!--
+    What do we need to consider when rolling out this change?
+-->
+
+Fixes #NNN <!-- This will close the issue automatically when you merge -->
+
+Completion workflow:
+
+- [ ] Ensure all pull request comments are addressed
+- [ ] Rebase your branch against main branch
+- [ ] Ensure that all checks passes with the latest changes on main branch
```

Comment:
```suggestion
- [ ] Ensure that all checks pass with the latest changes on main branch
```
Isn't this implicitly guarded against by not allowing merges when CI pipelines fail? In other words it can probably be removed?